### ISSUE TITLE
Can the gRPC opentelemetry plugin be split into another project?

### ISSUE THREAD

**Issue by john-forrest** [grpc/grpc#37817]
OK this is in the form of a question but essentially I am asking for a refactoring of how things are currently packaged so that the opentelemetry plugin (I see as cmake target gRPC::grpcpp_otel_plugin) is in a separate project. 

Why? Well as it currently stands, to build this plugin gRPC depends on opentelemetry-cpp. However, for opentelemetry-cpp to support otlp via grpc (build with WITH_OTLP_GRPC enabled), it in turn needs to pull in cmake package gRPC. This leads to a chicken and egg problem when it comes to building these modules. Perhaps I have a particular issue with this since we use the hunter C++ package manager, which by design does not support circular dependencies, but the scenario is not particularly elegant even if you can use brute force solutions. My current solution is to build opentelemetry-cpp with WITH_OTLP_GRPC=OFF, which removes the dependency but also the optimisation over using http.

To me the preferable scenario would either be to split this off to yet another module or perhaps to move this feature into opentelemetry-cpp itself. Question though: are there any technical reasons this can't happen - e.g. does the plugin depend on some internal/hidden features of gRPC itself?

**Comment by yashykt** [comment]
Yeah, this is a bit of an unfortunate thing, and you hit it right on the nail.

> are there any technical reasons this can't happen - e.g. does the plugin depend on some internal/hidden features of gRPC itself?

Yes, you are exactly right. This plugin reaches into the internals of gRPC. In the future, we might be able to make of the stuff that this plugin depends on public, but we are far away from that point currently.

So, till we reach that future, we have the following options - 
1) Use bazel. It doesn't have the chicken-egg problem that you are running into.

2) If you want to make it work with CMake and you want to use OTel C++'s OTLP exporter with gRPC enabled, this is what your build would look like - 
    * Build and install gRPC without the opentelemetry plugin (`gRPC_BUILD_GRPCPP_OTEL_PLUGIN` off)
    * Build opentelemetry c++ and the OTLP exporter
    * Build and install gRPC with the opentelemetry plugin

This is dirty I know, but I don't have a better solution at present :/
